compiler/rustc_borrowck/src/nll.rs

```diff
@@ -145,9 +145,11 @@ pub(crate) fn compute_regions<'a, 'tcx>(
         }
 
         if polonius_output {
-            let algorithm =
-                env::var("POLONIUS_ALGORITHM").unwrap_or_else(|_| String::from("Hybrid"));
-            let algorithm = Algorithm::from_str(&algorithm).unwrap();
+            let algorithm = infcx
+                .tcx
+                .env_var_os("POLONIUS_ALGORITHM".as_ref())
+                .unwrap_or_else(|| Symbol::intern("Hybrid"));
```

Comment:
I think a better solution here would be to add a `polonius_algorithm` field to `Options` and read the `POLONIUS_ALGORITHM` env var in `build_session_options`. Then marking the field as `[TRACKED]` is enough to get it tracked by incr comp.

Reply:
That is _exactly_ what I did originally in https://github.com/rust-lang/rust/pull/129342, but @petrochenkov [preferred](https://github.com/rust-lang/rust/pull/129342#issuecomment-2335153699) that we integrated it with the query system instead, hence this PR.

I have no idea what's the best solution here, I'll defer to you two to discuss and decide on that ;) .

Reply:
I did prefer accessing env vars as early as possible to prevent eg proc-macros from affecting compilation options. Proc-macros are known to do things they should not be allowed to do.

Reply:
Having a query for env vars probably still makes sense even if some of the env var uses are converted to options.

(One downside of early tracking is that the dependency on the env var will be registered even if the env var is not actually used.)